src/exoforce.py

```diff
@@ -57,103 +97,132 @@
 	print("This body's Id is: ", body_id)"""
 
 
-# COMMON SETUP
+# COMMON WORLD SETUP
 p.setGravity(0, 0, -9.81)
-# p.setTimeStep(-0.0001)
-p.setRealTimeSimulation(0) # XXX
+p.setRealTimeSimulation(0)  # Requirement for p.applyExternalForce() to work     # XXX
 color = [1, 0, 0]
 
-# IMPORTANT JOINT INDICES:
 
-"""
-00: Waist
-01: Base - between the feet (on the floor)
-04: Upper Chest
-07: Neck
-08: Head
-12: Right Shoulder
-13: Right Elbow
-16: Right Wrist
-21: Left Shoulder
-22: Left Elbow
-hand_link: Left Wrist
-30: Right Leg (Point where it joins the waist)
-31: Right Knee
-33: Right Foot
-38: Left Leh (Point where it joins the waist)
-39: Left Knee
-41: Left Foot
-"""
 
 
+######################################################################################
+###########                                                              #############
+###########    DEBUGLINES, PARAMS, CONSTRAINTS AND FORCE DEFINITIONS     #############
+###########                                                              #############
+######################################################################################
+
+
+# LINK POSITION DEFINITIONS:
+lefthand_pos = p.getLinkState(human, lefthand_link)[0]
+righthand_pos = p.getLinkState(human, righthand_link)[0]
+
+# TENDON ORIGIN DEFINITIONS:
```

Comment:
remove copy-paste and implement a Tendon class and move all relevant functions there 
```python
class Tendon:
       self.start_location = xx
       self.end_location =xx
       self.current_force = xx

       def __init__(self, start_location, end_location):
           ...

       def decompose_force(self, force):
           ...

       def apply_force(self, force):
           ...
       
       def visualize(self):
          ...
 
       def update(self, start, end, force):
          self.decompose_force()
          self.apply_force()
          self.visualize()
 
```

Reply:
the structure is proposed for guidance, you'll likely need additional params and member functions